### DISCUSSION TITLE
features: closeModal methods

### DISCUSSION THREAD

**Discussion by leo-web3** [rainbow-me/rainbowkit#1347] [Ideas]
What considerations might have led RainbowKit to not disclose their hook methods? This can be very limiting in custom development.

```
  const {
    closeModal: closeConnectModal,
    isModalOpen: connectModalOpen,
    openModal: openConnectModal,
  } = useModalStateValue();

  const {
    closeModal: closeAccountModal,
    isModalOpen: accountModalOpen,
    openModal: openAccountModal,
  } = useModalStateValue();

  const {
    closeModal: closeChainModal,
    isModalOpen: chainModalOpen,
    openModal: openChainModal,
  } = useModalStateValue();
```
  
  
  For the three modals below, I only see that the new version has opened up accountModalOpen and openAccountModal. Why won't they disclose the remaining closeModal methods?

**Comment by leo-web3** [comment]
When I want to change the method of closing the modal, from clicking to close to mouse out to close, I can't find a public method to close it. This restrictive approach makes development difficult

**Comment by DanielSinclair** [comment]
Hi @leo-web3. Can you share your usecase of what you're hoping to achieve with a close modal hook?

**Reply by matthew-t3** [reply]
We have a few situations where we want to programmatically close the modal ourselves, i.e. timeout

**Reply by 0xmkh** [reply]
If you have your modal open and you connect your wallet, we immediately close it. Would that be an issue in your case ?

**Reply by AmbitionsXXXV** [reply]
My current requirement is that I will encounter modal display errors and route jumps during the auth process. When the jump is over, I should actually close the modal directly, but it does not. At this time, I will need a closeModal method

**Reply by AmbitionsXXXV** [reply]
And under Custom Authentication, if the verification fails, you need to close the modal and reopen it to trigger the nonce request again. When the backend nonce interface is designed to be invalid after one use, that makes Custom Authentication will continue to fail if it fails once, which greatly affects the user experience.
